fuse_optimizers/src/batch_optimizer.cpp

```diff
@@ -47,21 +48,24 @@ namespace fuse_optimizers
 {
 
 BatchOptimizer::BatchOptimizer(
-  rclcpp::NodeOptions options,
+  fuse_core::node_interfaces::NodeInterfaces<ALL_FUSE_CORE_NODE_INTERFACES> interfaces,
   fuse_core::Graph::UniquePtr graph
 ):
-  fuse_optimizers::Optimizer(options, std::move(graph)),
+  fuse_optimizers::Optimizer(interfaces, std::move(graph)),
   combined_transaction_(fuse_core::Transaction::make_shared()),
   optimization_request_(false),
   start_time_(rclcpp::Time::max()),
   started_(false)
 {
-  params_.loadFromROS(private_node_handle);
+  params_.loadFromROS(interfaces_);
 
   // Configure a timer to trigger optimizations
-  optimize_timer_ = this->create_timer(
-    params_.optimization_period,
-    std::bind(&BatchOptimizer::optimizerTimerCallback, this)
+  optimize_timer_ = rclcpp::create_timer(
+    interfaces_,
+    clock_,
+    params_.optimization_period.to_chrono<std::chrono::nanoseconds>(),
```

Comment:
`rclcpp::create_timer()` takes an `rclcpp::Duration`, so no need to convert this. ba487fa